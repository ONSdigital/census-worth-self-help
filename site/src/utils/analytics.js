export default class Analytics {
  static trackEvent(category, action, name = "", value = "") {
    if (window._paq) {
      window._paq.push(["trackEvent", category, action, name, value])
    } else {
      console.debug(`Track event not set : ${category} - ${action} - ${name} - ${value}`)
    }
  }

  static trackSiteSearch(keyword, category, searchCount) {
    if (window._paq) {
      window._paq.push(["trackSiteSearch", keyword, category, searchCount])
    } else {
      console.debug(`Track site search : ${keyword} - ${category} - ${searchCount}`)
    }
  }

  static setPageType(pageType) {
    if (typeof window !== `undefined` && window._paq) {
      window._paq.push(["setCustomDimension", 1, pageType])
    }
  }
}