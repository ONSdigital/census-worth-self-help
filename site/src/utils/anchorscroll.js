export default (history, timeout = 1000) => {
  let observer
  let timeoutId

  const reset = () => {
    if (timeoutId) {
      clearTimeout(timeoutId)
      timeoutId = null
    }

    if (observer) {
      observer.disconnect()
    }
  }

  const convertTextToLink = text => {
    return text.toLowerCase().replace(/\s/g, "-")
  }

  const createScrollToElement = hash => {
    return () => {
      let headers = []
      let tags = ["h1", "h2", "h3"]

      tags.forEach(tag => {
        headers = headers.concat(Array.from(document.getElementsByTagName(tag)))
      })

      let element = headers.find(
        header => convertTextToLink(header.textContent) === hash
      )

      if (element) {
        element.scrollIntoView()
        reset()
        return true
      }

      return false
    }
  }

  function scroll(hash) {
    if (typeof hash !== "string") {
      return
    }

    const anchorText = hash.slice(1).toLowerCase()

    if (!anchorText) {
      return
    }

    const scrollToElement = createScrollToElement(anchorText)

    setTimeout(() => {
      if (scrollToElement()) {
        return
      }

      if (window.MutationObserver) {
        observer = new MutationObserver(scrollToElement)

        observer.observe(document, {
          attributes: true,
          childList: true,
          subtree: true
        })
      }

      timeoutId = setTimeout(reset, timeout)
    })
  }

  // listen for changes
  history.listen(({ location, action }) => {
    if (timeoutId) {
      reset()
    }

    if (action !== "POP") {
      return
    }

    if (location.hash) {
      scroll(location.hash)
    }
  })

  // also trigger on load
  if (history.location.hash) {
    scroll(history.location.hash)
  }
}
