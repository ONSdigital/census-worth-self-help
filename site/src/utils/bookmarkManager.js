export default class BookmarkManager {
  constructor() {
    this.bookmarks = JSON.parse(localStorage.getItem("bookmarks"))
    if (this.bookmarks === null) {
      this.bookmarks = []
    }
  }

  getTopBookmarks() {
    return this.bookmarks
  }

  isPageBookmarked(page) {
    return this.bookmarks.includes(page)
  }

  bookmarkPage(page) {
    if (!this.isPageBookmarked(page)) {
      this.bookmarks.push(page)
    }
    localStorage.setItem("bookmarks", JSON.stringify(this.bookmarks))
  }

  unBookmarkPage(page) {
    this.bookmarks = this.bookmarks.filter(function(bookmark) {
      return bookmark !== page
    })

    localStorage.setItem("bookmarks", JSON.stringify(this.bookmarks))
  }
}
