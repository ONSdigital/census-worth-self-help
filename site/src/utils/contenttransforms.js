var sanitizeHtml = require("sanitize-html")

export const transformSources = htmlString => {
  let videoPath = process.env.GATSBY_ASSETS_PATH
    ? process.env.GATSBY_ASSETS_PATH
    : ""
  if (!htmlString) {
    return ""
  }
  return htmlString.replace(/{{TARGET_ASSETS_SRC}}/g, videoPath)
}

export const htmlSanitize = htmlString => {
  return sanitizeHtml(htmlString, {
    allowedTags: sanitizeHtml.defaults.allowedTags.filter(function(tag) {
      return tag !== "iframe"
    }).concat([
      "video",
      "audio",
      "img",
      "source",
      "h1",
      "h2"
    ]),
    allowedAttributes: {
      a: ["href", "name", "target"],
      img: ["src", "title"],
      source: ["src", "type"],
      video: ["controls"],
      audio: ["controls"]
    }
  })
}
