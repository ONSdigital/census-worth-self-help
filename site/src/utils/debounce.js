import _ from "lodash"

// debounce with default delay, or return sync function if delay exactly zero (for unit tests)
export default (fn, delay) => {
  return (delay === 0)
    ? fn
    : _.debounce((...args) => fn(...args), delay || 1000)
}