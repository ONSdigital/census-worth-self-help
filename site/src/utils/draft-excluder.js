const READY_FOR_PRODUCTION = "Ready for Live Site"

// Articles in draft should not be deployed to production. An article is draft if the draftreason for the article
// is anything other than "Ready for production" - see ONS-359.
//
// We also use this Draft Excluder concept to not show articles that are for the call centre only.
//
// Initially this was done using the draft tag. This technique might be removed in the future.
exports.excludeDraftArticle = node => {
  return Boolean(
    process.env.EXCLUDE_DRAFTS && (
      (
        node.frontmatter.tags &&
        node.frontmatter.tags.includes("draft")
      ) || (
        node.frontmatter.draftreason &&
        node.frontmatter.draftreason !== READY_FOR_PRODUCTION
      ) || node.frontmatter.cconly
    )
  )
}
