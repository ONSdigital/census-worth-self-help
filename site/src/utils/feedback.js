export default class Feedback {
  static bookmarkClickEvent(article) {
    Feedback.feedbackEvent("click-event", "bookmark", article)
  }
  static articleWasBookmarked(article) {
    Feedback.feedbackEvent("article-was-bookmarked", "bookmarked", article, 1)
  }
  static articleWasUnBookmarked(article) {
    Feedback.feedbackEvent("article-was-bookmarked", "unbookmarked", article, 1)
  }
  static articleIsUseful(article, reason) {
    Feedback.feedbackEvent(
        "article-feedback-review",
        "POSITIVE REVIEW: " + article,
        reason
    )
    Feedback.feedbackEvent("article-feedback-rating", "rating", article, 1)
  }
  static articleIsNotUseful(article, reason) {
    Feedback.feedbackEvent(
      "article-feedback-review",
      "REVIEW: " + article,
      reason
    )
    Feedback.feedbackEvent("article-feedback-rating", "rating", article, -1)
  }
  static feedbackEvent(category, action, name = "", value = "") {
    if (window._paq) {
      window._paq.push(["trackEvent", category, action, name, value])
    }
  }
}
