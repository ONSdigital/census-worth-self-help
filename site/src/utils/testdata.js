export const data = {
  edges: [
    {
      node: {
        frontmatter: {
          title: "test Article 1",
          description: "testDescription",
          priority: 1,
          date: new Date()
        },
        fields: {
          pagename: "test-article-1",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Article 2",
          description: "testDescription",
          priority: 0,
          date: new Date(),
          tags: ["popular", "webchat"]
        },
        fields: {
          pagename: "test-article-2",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Article 3",
          description: "testDescription",
          priority: 0,
          date: new Date(),
          tags: ["popular"]
        },
        fields: {
          pagename: "test-article-3",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Article 4",
          description: "testDescription",
          priority: 0,
          date: new Date(),
          tags: ["draft"]
        },
        fields: {
          pagename: "test-article-4",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Article 5",
          description: "testDescription",
          priority: 0,
          date: new Date(),
          tags: ["popular"]
        },
        fields: {
          pagename: "test-article-5",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Article 6",
          description: "testDescription",
          priority: 0,
          date: new Date(),
          tags: ["popular"]
        },
        fields: {
          pagename: "test-article-6",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Directory 7",
          description: "testDescription",
          priority: 0,
          date: new Date()
        },
        fields: {
          pagename: "test-directory-1",
          collection: "directories"
        }
      }
    }
  ]
}

export const articleNode = data.edges[0].node
export const webchatArticleNode = data.edges[1].node
export const directoryNode = data.edges[2].node
export const draftNode = data.edges[3].node
export const popularList = {
  edges: [data.edges[1], data.edges[2], data.edges[4], data.edges[5]]
}
export const articleList = { edges: data.edges.slice(0, 6) }
