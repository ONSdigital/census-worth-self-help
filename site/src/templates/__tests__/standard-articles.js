import React from "react"
import renderer from "react-test-renderer"

import StandardArticle from "../standard-article"

import {articleNode, articleList} from "../../utils/testdata"

describe("StandardArticle", () => {
  const pageContext = {
  	breadcrumbs : [],
  	peers : [ {title: "nothing"}, {title: "test Article 2"}],
  	title : "test Article 1"
  }
  const data = {
  	markdownRemark : articleNode,
    allMarkdownRemark : articleList
  }
  it("renders correctly", () => {
    const tree = renderer
      .create(<StandardArticle data={data} pageContext={pageContext} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})