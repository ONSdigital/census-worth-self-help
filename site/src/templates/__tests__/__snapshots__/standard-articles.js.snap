// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StandardArticle renders correctly 1`] = `
<div
  css={
    Object {
      "map": undefined,
      "name": "vkixka",
      "next": undefined,
      "styles": "
        height: 100vh;
        display: flex;
        flex-direction: column;
      ",
    }
  }
>
  <header
    css={
      Object {
        "map": undefined,
        "name": "tejz0l",
        "next": undefined,
        "styles": "
        width: 100vw;
        
    background: linear-gradient(to left, #003d59, #33637a);
  ;
      ",
      }
    }
  >
    <div
      css={
        Object {
          "map": undefined,
          "name": "1thgah9",
          "next": undefined,
          "styles": "
            display: flex;
          ",
        }
      }
    >
      <div
        className="Header-Title-Style"
        css={
          Object {
            "map": undefined,
            "name": "17ih1bc",
            "next": undefined,
            "styles": "
              
    display: flex;
    justify-content: center;
    flex-direction: column;
  
              
    padding: 15px;
  
              flex-grow: 1;
            ",
          }
        }
        onClick={[Function]}
      >
        Census Field Assistant
      </div>
      <div
        className="Header-button-Style"
        css={
          Object {
            "map": undefined,
            "name": "1sm5jo",
            "next": undefined,
            "styles": "
        
    display: flex;
    justify-content: center;
    flex-direction: column;
  
        padding: 0px 10px;
        background-color: rgb(255, 255, 255, 0.2);
        border-radius: 4px;
        margin: 10px 10px 10px 0px;
        min-width: 20px;
        object-fit: contain;
        text-align: center;
        height: 40px;
      ",
          }
        }
        data-testid="topbarlink-button"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-search fa-w-16 "
          data-icon="search"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        className="Header-button-Style"
        css={
          Object {
            "map": undefined,
            "name": "1sm5jo",
            "next": undefined,
            "styles": "
        
    display: flex;
    justify-content: center;
    flex-direction: column;
  
        padding: 0px 10px;
        background-color: rgb(255, 255, 255, 0.2);
        border-radius: 4px;
        margin: 10px 10px 10px 0px;
        min-width: 20px;
        object-fit: contain;
        text-align: center;
        height: 40px;
      ",
          }
        }
        data-testid="topbarlink-button"
        onClick={[Function]}
      >
        Menu
      </div>
    </div>
  </header>
  <div
    css={
      Object {
        "map": undefined,
        "name": "o9ztju",
        "next": undefined,
        "styles": "
          
    margin: 0 auto;
    max-width: 700px;
  
          overflow: scroll;
          width: 100vw;
          display: flex;
          flex-direction: column;
          flex-grow: 1;
        ",
      }
    }
  >
    <div
      css={
        Object {
          "map": undefined,
          "name": "1k0aln6",
          "next": undefined,
          "styles": "
            flex-grow: 1;
          ",
        }
      }
    >
      <div
        css={
          Object {
            "map": undefined,
            "name": "ypl6yy",
            "next": undefined,
            "styles": "
        border-bottom: 1px rgb(229, 229, 234) solid;
        padding: 15px 0px;
      ",
          }
        }
      >
        <div
          className="Section-heading-Style"
          css={
            Object {
              "map": undefined,
              "name": "136nmlb",
              "next": undefined,
              "styles": "
          
    padding: 5px 15px;
  
        ",
            }
          }
        >
          EXPLORE CONTENT
        </div>
        <div
          css={
            Object {
              "map": undefined,
              "name": "136nmlb",
              "next": undefined,
              "styles": "
          
    padding: 5px 15px;
  
        ",
            }
          }
        >
          <div
            className="Button-subhead-Style css-2b097c-container"
            onKeyDown={[Function]}
          >
            <div
              className=" css-yk16xz-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className=" css-1hwfws3"
              >
                <div
                  className=" css-1wa3eu0-placeholder"
                >
                  Select...
                </div>
                <div
                  className="css-1g6gooi"
                >
                  <div
                    className=""
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-autocomplete="list"
                      autoCapitalize="none"
                      autoComplete="off"
                      autoCorrect="off"
                      disabled={false}
                      id="react-select-2-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      spellCheck="false"
                      style={
                        Object {
                          "background": 0,
                          "border": 0,
                          "boxSizing": "content-box",
                          "color": "inherit",
                          "fontSize": "inherit",
                          "label": "input",
                          "opacity": 1,
                          "outline": 0,
                          "padding": 0,
                          "width": "1px",
                        }
                      }
                      tabIndex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </div>
              </div>
              <div
                className=" css-1wy0on6"
              >
                <div
                  aria-hidden="true"
                  className=" css-tlfecz-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="css-19bqh2r"
                    focusable="false"
                    height={20}
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        css={
          Object {
            "map": undefined,
            "name": "bj0o6b",
            "next": undefined,
            "styles": "
            background-color: white;
          ",
          }
        }
      >
        <div
          css={
            Object {
              "map": undefined,
              "name": "12ew49b",
              "next": undefined,
              "styles": "
        
    padding: 5px 15px;
  
      ",
            }
          }
        >
          <h1
            className="Article-Title-Style"
            css={
              Object {
                "map": undefined,
                "name": "1xcgh0k",
                "next": undefined,
                "styles": "
          margin-bottom: 0px;
          display: flex;
        ",
              }
            }
            data-testid="search-result-title"
          >
            test Article 1
          </h1>
          <div
            className="Button-subhead-Style"
            css={
              Object {
                "map": undefined,
                "name": "wzn8if",
                "next": undefined,
                "styles": "
            padding-top: 5px;
            padding-bottom: 15px;
          ",
              }
            }
          >
            <span>
              Last updated:
              <i>
                 a few seconds ago
              </i>
            </span>
          </div>
        </div>
        <div
          css={
            Object {
              "map": undefined,
              "name": "1vf6ctb",
              "next": undefined,
              "styles": "
        border-radius: 4px;
        box-shadow: 0 2px 4px 0 rgba(212, 212, 212, 0.5);
        border: solid 1px #ececed;
        margin: 0px 10px;
        background-color: rgb(255, 255, 255);
        display: flex;
      ",
            }
          }
          data-testid="block-button"
          onClick={[Function]}
        >
          <div
            className="Article-Title-Style"
            css={
              Object {
                "map": undefined,
                "name": "14yivzb",
                "next": undefined,
                "styles": "
            
    display: flex;
    justify-content: center;
    flex-direction: column;
  
            display: flex;
            justify-content: center;
            height: 78px;
            width: 78px;
            color: white;
            background-color: #003d59;
          ",
              }
            }
          >
            <div
              css={
                Object {
                  "map": undefined,
                  "name": "1n8pnzb",
                  "next": undefined,
                  "styles": "
              text-align: center;
            ",
                }
              }
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-bookmark fa-w-12 "
                data-icon="bookmark"
                data-prefix="far"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M336 0H48C21.49 0 0 21.49 0 48v464l192-112 192 112V48c0-26.51-21.49-48-48-48zm0 428.43l-144-84-144 84V54a6 6 0 0 1 6-6h276c3.314 0 6 2.683 6 5.996V428.43z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
          <div
            css={
              Object {
                "map": undefined,
                "name": "i8tjei",
                "next": undefined,
                "styles": "
          
    display: flex;
    justify-content: center;
    flex-direction: column;
  
          margin: 0px 20px;
          color: #003d59;
        ",
              }
            }
          >
            <div
              className="Button-heading-Style"
              css={
                Object {
                  "map": undefined,
                  "name": "km4e7l",
                  "next": undefined,
                  "styles": "
            text-align: left;
          ",
                }
              }
            >
              Bookmark this page
            </div>
            <div
              className="Card-meta-Style"
              css={
                Object {
                  "map": undefined,
                  "name": "km4e7l",
                  "next": undefined,
                  "styles": "
            text-align: left;
          ",
                }
              }
            >
              Save it to view later
            </div>
          </div>
        </div>
        <div
          css={
            Object {
              "map": undefined,
              "name": "a00x3a",
              "next": undefined,
              "styles": "
              margin-top: 20px;
            ",
            }
          }
        >
          <div
            className="Article-body-Style"
            css={
              Object {
                "map": undefined,
                "name": "pywoey",
                "next": undefined,
                "styles": "
        
    padding: 5px 15px;
  ;
      ",
              }
            }
          >
            <div
              className="Article-sub-title-Style"
            >
              testDescription
            </div>
            <div
              className="article-content"
              dangerouslySetInnerHTML={
                Object {
                  "__html": undefined,
                }
              }
            />
          </div>
        </div>
        <div
          css={
            Object {
              "map": undefined,
              "name": "ypl6yy",
              "next": undefined,
              "styles": "
        border-bottom: 1px rgb(229, 229, 234) solid;
        padding: 15px 0px;
      ",
            }
          }
        >
          <div
            className="Section-heading-Style"
            css={
              Object {
                "map": undefined,
                "name": "1mp9jtn",
                "next": undefined,
                "styles": "
                display: flex;
                
    padding: 15px 0px;
  
                
    padding: 5px 15px;
  
              ",
              }
            }
          >
            HOW WOULD YOU RATE THIS CONTENT?
          </div>
          <div
            css={
              Object {
                "map": undefined,
                "name": "aoc9mf",
                "next": undefined,
                "styles": "
                display: flex;
              ",
              }
            }
          >
            <div
              css={
                Object {
                  "map": undefined,
                  "name": "ltouev",
                  "next": undefined,
                  "styles": "
        
    display: flex;
    justify-content: center;
    flex-direction: column;
  
        padding: 0px 10px;
        border-radius: 4px;
        box-shadow: 0 2px 4px 0 rgba(212, 212, 212, 0.5);
        border: solid 1px #ececed;
        margin: 0px 10px;
        text-align: center;
        height: 40px;
        background-color: rgb(255, 255, 255);
        
                  flex: 1;
                  margin-right: 5px;
                
      ",
                }
              }
              data-testid="large-button"
              onClick={[Function]}
            >
              <div
                css={
                  Object {
                    "map": undefined,
                    "name": "1qs2efx",
                    "next": undefined,
                    "styles": "
            position: relative;
            color: #003d59;
          ",
                  }
                }
              >
                <div
                  css={
                    Object {
                      "map": undefined,
                      "name": "1wo7hj",
                      "next": undefined,
                      "styles": "
              position: absolute;
            ",
                    }
                  }
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-thumbs-up fa-w-16 "
                    data-icon="thumbs-up"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M104 224H24c-13.255 0-24 10.745-24 24v240c0 13.255 10.745 24 24 24h80c13.255 0 24-10.745 24-24V248c0-13.255-10.745-24-24-24zM64 472c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24zM384 81.452c0 42.416-25.97 66.208-33.277 94.548h101.723c33.397 0 59.397 27.746 59.553 58.098.084 17.938-7.546 37.249-19.439 49.197l-.11.11c9.836 23.337 8.237 56.037-9.308 79.469 8.681 25.895-.069 57.704-16.382 74.757 4.298 17.598 2.244 32.575-6.148 44.632C440.202 511.587 389.616 512 346.839 512l-2.845-.001c-48.287-.017-87.806-17.598-119.56-31.725-15.957-7.099-36.821-15.887-52.651-16.178-6.54-.12-11.783-5.457-11.783-11.998v-213.77c0-3.2 1.282-6.271 3.558-8.521 39.614-39.144 56.648-80.587 89.117-113.111 14.804-14.832 20.188-37.236 25.393-58.902C282.515 39.293 291.817 0 312 0c24 0 72 8 72 81.452z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </div>
              </div>
              <div
                className="Button-heading-Style"
                css={
                  Object {
                    "map": undefined,
                    "name": "1bt83uw",
                    "next": undefined,
                    "styles": "
          margin: 0px 20px;
          color: #003d59;
        ",
                  }
                }
              >
                Useful
              </div>
            </div>
            <div
              css={
                Object {
                  "map": undefined,
                  "name": "122bqh2",
                  "next": undefined,
                  "styles": "
        
    display: flex;
    justify-content: center;
    flex-direction: column;
  
        padding: 0px 10px;
        border-radius: 4px;
        box-shadow: 0 2px 4px 0 rgba(212, 212, 212, 0.5);
        border: solid 1px #ececed;
        margin: 0px 10px;
        text-align: center;
        height: 40px;
        background-color: rgb(255, 255, 255);
        
                  flex: 1;
                  margin-left: 5px;
                
      ",
                }
              }
              data-testid="large-button"
              onClick={[Function]}
            >
              <div
                css={
                  Object {
                    "map": undefined,
                    "name": "1qs2efx",
                    "next": undefined,
                    "styles": "
            position: relative;
            color: #003d59;
          ",
                  }
                }
              >
                <div
                  css={
                    Object {
                      "map": undefined,
                      "name": "1wo7hj",
                      "next": undefined,
                      "styles": "
              position: absolute;
            ",
                    }
                  }
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-thumbs-down fa-w-16 "
                    data-icon="thumbs-down"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 56v240c0 13.255 10.745 24 24 24h80c13.255 0 24-10.745 24-24V56c0-13.255-10.745-24-24-24H24C10.745 32 0 42.745 0 56zm40 200c0-13.255 10.745-24 24-24s24 10.745 24 24-10.745 24-24 24-24-10.745-24-24zm272 256c-20.183 0-29.485-39.293-33.931-57.795-5.206-21.666-10.589-44.07-25.393-58.902-32.469-32.524-49.503-73.967-89.117-113.111a11.98 11.98 0 0 1-3.558-8.521V59.901c0-6.541 5.243-11.878 11.783-11.998 15.831-.29 36.694-9.079 52.651-16.178C256.189 17.598 295.709.017 343.995 0h2.844c42.777 0 93.363.413 113.774 29.737 8.392 12.057 10.446 27.034 6.148 44.632 16.312 17.053 25.063 48.863 16.382 74.757 17.544 23.432 19.143 56.132 9.308 79.469l.11.11c11.893 11.949 19.523 31.259 19.439 49.197-.156 30.352-26.157 58.098-59.553 58.098H350.723C358.03 364.34 384 388.132 384 430.548 384 504 336 512 312 512z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </div>
              </div>
              <div
                className="Button-heading-Style"
                css={
                  Object {
                    "map": undefined,
                    "name": "1bt83uw",
                    "next": undefined,
                    "styles": "
          margin: 0px 20px;
          color: #003d59;
        ",
                  }
                }
              >
                Not useful
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          css={
            Object {
              "map": undefined,
              "name": "ypl6yy",
              "next": undefined,
              "styles": "
        border-bottom: 1px rgb(229, 229, 234) solid;
        padding: 15px 0px;
      ",
            }
          }
        >
          <div
            css={
              Object {
                "map": undefined,
                "name": "1p7jpnp",
                "next": undefined,
                "styles": "
        
    padding: 15px 0px;
  ;
      ",
              }
            }
          >
            <div
              css={
                Object {
                  "map": undefined,
                  "name": "qvcny6",
                  "next": undefined,
                  "styles": "
          
    padding: 15px 0px;
  ;
          
    padding: 5px 15px;
  
          display: flex;
        ",
                }
              }
            >
              <div
                className="Section-heading-Style"
              >
                ALSO IN THIS TOPIC
              </div>
            </div>
            <div
              css={
                Object {
                  "map": undefined,
                  "name": "1mij31f",
                  "next": undefined,
                  "styles": "
        
    margin: 2px 0px;
  ;
      ",
                }
              }
            >
              <div
                css={
                  Object {
                    "map": undefined,
                    "name": "1q53lw7",
                    "next": undefined,
                    "styles": "
        
    padding: 15px;
    min-height: 30px;
  ;
        box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.5);
        border-left: 6px solid #003d59;
        background-color: rgb(255, 255, 255);
      ",
                  }
                }
                onClick={[Function]}
              >
                <div
                  className="Card-heading-Style"
                  css={
                    Object {
                      "map": undefined,
                      "name": "qiy98z",
                      "next": undefined,
                      "styles": "
          color: inherit;
          flex-grow: 1;
          font-weight: semi-bold;
        ",
                    }
                  }
                  data-testid="articletab-article-card"
                >
                  test Article 2
                </div>
                <div
                  className="Card-sub-head-Style-gray"
                  css={
                    Object {
                      "map": undefined,
                      "name": "1hejrb1",
                      "next": undefined,
                      "styles": "
          padding-top: 6px;
          display: flex;
          flex-direction: vertical;
          line-height: 1.08;
          max-height: 42.12px; // line-height * font-size * 3
        ",
                    }
                  }
                >
                  <div
                    css={
                      Object {
                        "map": undefined,
                        "name": "lz6rtp",
                        "next": undefined,
                        "styles": "
            width: 70%;
            overflow: hidden;
            text-overflow: ellipsis;
            flex-grow: 1;
          ",
                      }
                    }
                  >
                    testDescription
                  </div>
                  <div
                    className="Card-meta-Style"
                    css={
                      Object {
                        "map": undefined,
                        "name": "4j3tb8",
                        "next": undefined,
                        "styles": "
            display: flex;
            flex-direction: column;
            justify-content: flex-end;
          ",
                      }
                    }
                  >
                    a few seconds ago
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      css={
        Object {
          "map": undefined,
          "name": "ej75ms",
          "next": undefined,
          "styles": "
        
    padding: 30px 0px;
  
        width: 100%;
        padding: 10px 0px;
        background-color: rgb(236, 236, 237);
        display: flex;
        min-height: 100px;
        flex-direction: column;
        justify-content: center;
        flex-shrink: 0;
      ",
        }
      }
    >
      <img
        alt="Footer logo"
        css={
          Object {
            "map": undefined,
            "name": "1v6eegl",
            "next": undefined,
            "styles": "
          margin-bottom: 50px;
          margin-top: 50px;
        ",
          }
        }
        src="/footer_logo.svg"
      />
      <div
        css={
          Object {
            "map": undefined,
            "name": "buygrl",
            "next": undefined,
            "styles": "
            margin-bottom: 34px;
          ",
          }
        }
      >
        <div
          css={
            Object {
              "map": undefined,
              "name": "1k8b10p",
              "next": undefined,
              "styles": "
        
    display: flex;
    justify-content: center;
    flex-direction: column;
  
        padding: 0px 10px;
        border-radius: 4px;
        box-shadow: 0 2px 4px 0 rgba(212, 212, 212, 0.5);
        border: solid 1px #ececed;
        margin: 0px 10px;
        text-align: center;
        height: 40px;
        background-color: rgb(255, 255, 255);
        
      ",
            }
          }
          data-testid="large-button"
          onClick={[Function]}
        >
          <div
            css={
              Object {
                "map": undefined,
                "name": "1qs2efx",
                "next": undefined,
                "styles": "
            position: relative;
            color: #003d59;
          ",
              }
            }
          >
            <div
              css={
                Object {
                  "map": undefined,
                  "name": "1wo7hj",
                  "next": undefined,
                  "styles": "
              position: absolute;
            ",
                }
              }
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-phone fa-w-16 "
                data-icon="phone"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
          <div
            className="Button-heading-Style"
            css={
              Object {
                "map": undefined,
                "name": "1bt83uw",
                "next": undefined,
                "styles": "
          margin: 0px 20px;
          color: #003d59;
        ",
              }
            }
          >
            Census Field Support
          </div>
        </div>
      </div>
      <div
        className="Footer-Style"
      >
        © Office for National Statistics 2019
      </div>
    </div>
  </div>
</div>
`;
